x-pack/plugins/data_enhanced/server/search/session/session_service.ts

```diff
@@ -325,8 +296,22 @@ export class SearchSessionService implements ISessionService {
     );
   };
 
+  public extend(deps: SearchSessionDependencies, sessionId: string, keepAlive: string) {
+    this.logger.debug(`extend | ${sessionId}`);
+
+    // Calculate the new `expires` value given the `keepAlive`
+    const expiresMoment = dateMath.parse(`now+${keepAlive}`);
+    if (!expiresMoment || isNaN(expiresMoment.date())) {
+      throw new Error(`"${keepAlive}" is not a valid value for keepAlive`);
+    }
+    const expires = expiresMoment.toISOString();
+
+    // Update the `expires` value in the search session saved object
+    return this.update(deps, sessionId, { expires });
```

Comment:
This still doesn't update the individual search requests, right?

Reply:
The session service itself only updates the session object. But the `extendSession` exposed by the search service calls this method, in addition to extending each search request.

Reply:
![75231797_2342379812537974_2815010140908945408_n](https://user-images.githubusercontent.com/3016806/104589133-185de800-5672-11eb-9f80-0d954536b715.jpg)

Reply:
Just a thought - 
Maybe exposing the session service on the public contract is no longer necessary?
The methods on it don't really make sense outside the context of the search service.

Reply:
I'm not totally sure I understand. Are you referring to what is returned by `asScoped`? Right now all of the methods it exposes are used (with the exception of `updateSession`, which I assume will probably not be needed, since we will probably introduce a dedicated `renameSession` or something like that).

Reply:
@lukasolson I still don't understand something.
When updating the session, nothing calls the actual `async_search` api to update the expiration.